source/agora/node/Ledger.d

```diff
@@ -165,7 +165,19 @@ public class Ledger
                 "different from the one in the config file");
 
         if (this.utxo_set.length == 0)
-            this.rebuildUTXOAndValidatorSet(gen_block);
+        {
+            // clear validator set
+            this.enroll_man.removeAllValidators();
+
+            foreach (height; 0 .. this.last_block.header.height + 1)
+            {
+                Block block;
+                this.storage.readBlock(block, Height(height));
+                if (!this.acceptBlock(block, false))
+                    break;
```

Comment:
This looks like dead code to me. With `externalize = false` `acceptBlock` will throw instead of returning false, right?

Reply:
> This looks like dead code to me. With externalize = false, acceptBlock will throw instead of returning false, right?

Right, will update.